internal/pkg/util/fs/overlay/overlay_linux.go

```diff
@@ -133,3 +140,71 @@ func IsIncompatible(err error) bool {
 	}
 	return false
 }
+
+var ErrNoRootlessOverlay = errors.New("rootless overlay not supported by kernel")
+
+// CheckRootless checks whether the kernel overlay driver supports unprivileged use in a user namespace.
+func CheckRootless() error {
+	mountBin, err := bin.FindBin("mount")
+	if err != nil {
+		return fmt.Errorf("while looking for mount command: %s", err)
+	}
+
+	tmpDir, err := os.MkdirTemp("", "check-overlay")
+	if err != nil {
+		return fmt.Errorf("while creating temporary directories: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	lowerDir := filepath.Join(tmpDir, "l")
+	if err := os.Mkdir(lowerDir, 0o777); err != nil {
+		return fmt.Errorf("while creating temporary directories: %w", err)
+	}
+	upperDir := filepath.Join(tmpDir, "u")
+	if err := os.Mkdir(upperDir, 0o777); err != nil {
+		return fmt.Errorf("while creating temporary directories: %w", err)
+	}
+	workDir := filepath.Join(tmpDir, "w")
+	if err := os.Mkdir(workDir, 0o777); err != nil {
+		return fmt.Errorf("while creating temporary directories: %w", err)
+	}
+	mountDir := filepath.Join(tmpDir, "m")
+	if err := os.Mkdir(mountDir, 0o777); err != nil {
+		return fmt.Errorf("while creating temporary directories: %w", err)
+	}
+
+	args := []string{
+		"-t", "overlay",
+		"-o", fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s,userxattr", lowerDir, upperDir, workDir),
+		"none",
+		mountDir,
+	}
+
```

Comment:
@cclerget pointed out to me via Slack that this can be simplified.  For the "-o" option use the constant "lowerdir=/tmp:/etc" and replace "mountDir" with "/tmp".   Then remove all references to "tmpDir".

Reply:
It turns out that the simplified version doesn't work at least on Ubuntu 22.04.  For some reason it gives an error
```
mount: /tmp: wrong fs type, bad option, bad superblock on none, missing codepage or helper program, or other error.
```
Replacing the first "/tmp" with "/mnt" makes it work.